networkx/algorithms/sampling.py

```diff
@@ -0,0 +1,278 @@
+# -*- encoding: utf-8 -*-
+# sampling.py - functions for sampling nodes from a graph
+#
+# Copyright 2011 Maciej Kurant.
+# Copyright 2015 NetworkX developers.
+#
+# This file is part of NetworkX.
+#
+# NetworkX is distributed under a BSD license; see LICENSE.txt for more
+# information.
+"""Provides functions for sampling nodes from a graph."""
+from __future__ import division
+
+import bisect
+import itertools
+import random
+
+import networkx as nx
+
+__all__ = ['degree_weighted_independent_node_sample', 'random_walk',
+           'uniform_independent_node_sample']
+
+
+# def estimate_mean(sample, values, weights=None):
```

Comment:
I question whether this should function should be provided by NetworkX. This function has nothing in particular to do with graphs, so it should not be provided as part of the public API of NetworkX. At best, these should be provided as examples only.